src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionRequestHandler.cs

```diff
@@ -391,6 +391,14 @@ private void HandleAfterTestRunEnd(Message message)
 
             var attachmentsets = this.dataCollectionManager.SessionEnded(isCancelled);
             var invokedDataCollectors = this.dataCollectionManager.GetInvokedDataCollectors();
+            if (invokedDataCollectors != null && invokedDataCollectors.Count > 0)
+            {
+                foreach (var invokedDataCollector in invokedDataCollectors)
```

Comment:
@AbhitejJohn this is the place where we want to send an "object metric"

cc: @jakubch1